assignment/assignment.hs

```diff
@@ -26,24 +31,44 @@ import Data.Char(intToDigit)
 -- predicate returns `True`. Implement `myfilter` using `foldl`. (Hint: you 
 -- might need to use `++` rather than `:` to build the result)
 
--- myfilter :: (a -> Bool) -> [a] -> [a]
+--conditionalAttach creates functions with 
+conditionalAttach :: (a -> Bool) -> [a] -> a -> [a]
+conditionalAttach condition lis a = 
+  if(condition a)
+    then lis ++ [a] 
+    else lis
+myfilter :: (a -> Bool) -> [a] -> [a]
+myfilter check items = foldl (conditionalAttach check) [] items 
 
 
 -------
 -- 3 --
 -------
 -- The `and` and `or` functions take a list of booleans.
--- `and` returns true only iff all the values are true.
+-- `and` returns true only if all the values are true.
 -- `or` returns true if any are true.
 --
 -- (Side note: when implementing your functions, did you get a wavy
 -- line suggestion say that you could replace your code with the
 -- library function. Stop for a minute and see if you can work
 -- out how Haskell worked this out.)
-  
--- myand:: [ Bool ] -> Bool 
 
--- myor:: [ Bool ] -> Bool 
+stillTrue :: Bool -> Bool -> Bool
+stillTrue old curr = 
+  if(old && curr)
+    then True
+    else False
+myand:: [ Bool ] -> Bool
+myand bools = foldl stillTrue True bools   
+
+anyTrue :: Bool -> Bool -> Bool
+anyTrue old curr =
+  if(old || curr)
+    then True
+    else False
+
+myor:: [ Bool ] -> Bool
+myor bools = foldl anyTrue False bools
 
 
```

Comment:
If you think of folks as a kind of PacMan consuming the list, then `and`  means "everything consumed so far it true and the result of applying and to the rest if the list is true.

This means you can write it as

~~~ hs
 foldl (\l h -> h:l) []
~~